tests/python/test_field.py

```diff
@@ -146,3 +146,36 @@ def test_field_name():
     for i in range(10):
         d.append(ti.field(dtype=ti.f32, shape=(2, 3), name=f'd{i}'))
         assert d[i].name == f'd{i}'
+
+
+@ti.test()
+@pytest.mark.parametrize('shape', field_shapes)
+def test_field_copy_from(shape):
+    shapes = [ti.i32, ti.f32]  # Metal kernel only supports <= 32-bit data
```

Comment:
shape doesn't match what's inside the array :P 
Using parameterize for dtype should work and pass CI  :D 
```
+@ti.test()
+@pytest.mark.parametrize('shape', field_shapes)
+@pytest.mark.parametrize('dtype', [ti.i32, ti.f32])
+def test_field_copy_from(shape, dtype):
+    x = ti.field(dtype=ti.f32, shape=shape)
+    other = ti.field(dtype=dtype, shape=shape)
+    other.fill(1)
+    x.copy_from(other)
+    convert = lambda arr: arr[0] if len(arr) == 1 else arr
+    assert (convert(x.shape) == shape)
+    assert (x.dtype == ti.f32)
+    assert ((x.to_numpy() == 1).all())
```

Reply:
@ailzhang 
Hi, I finally have time to keep working on this PR, how can I test locally? I have run the pytest script.